quarkus/tests/integration/src/test/java/org/keycloak/it/cli/dist/JaxRsDistTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.it.cli.dist;
+
+import org.awaitility.Awaitility;
+import org.junit.jupiter.api.Test;
+import org.keycloak.it.jaxrs.filter.TestFilterTestProvider;
+import org.keycloak.it.junit5.extension.CLIResult;
+import org.keycloak.it.junit5.extension.DistributionTest;
+import org.keycloak.it.junit5.extension.RawDistOnly;
+import org.keycloak.it.junit5.extension.TestProvider;
+import org.keycloak.it.utils.KeycloakDistribution;
+
+import java.util.concurrent.TimeUnit;
+
+import static io.restassured.RestAssured.when;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@DistributionTest(keepAlive = true)
+@RawDistOnly(reason = "Containers are immutable")
+public class JaxRsDistTest {
+
+    @Test
+    @TestProvider(TestFilterTestProvider.class)
+    public void requestFilterTest(KeycloakDistribution dist) {
+        CLIResult cliResult = dist.run("start-dev");
+
+        cliResult.assertStartedDevMode();
+
+        assertEquals(200, when().get("/").getStatusCode());
+
+        Awaitility.await().atMost(5, TimeUnit.SECONDS).untilAsserted(
+                () -> cliResult.assertMessage("Request GET / has context request true has keycloaksession false"));
```

Comment:
Shouldn't keycloaksession be true as well?

Reply:
That's the behavior I'm seen with the live server as well - hitting just the landing page seems to happen before the handler chain is invoked and the session is associated. There's a related issue with this in that when you use prematching and resteasy reactive filters the session is always null.

Reply:
To expand on this, with the filter annotated with PreMatching we see:

```
Request GET /realms/master/ has context request true has keycloaksession false
Request GET / has context request true has keycloaksession false
```

Without PreMatching, /realms/master/ now has a session:
```
Request GET /realms/master/ has context request true has keycloaksession true
Request GET / has context request true has keycloaksession false
```

This mimicks the behavior seen when using resteasy reactive filters from https://github.com/keycloak/keycloak/issues/25038#issuecomment-1966464394

The user indicated the PreMatching in prior releases of Keycloak still provided access to the session, so it does appear to be a difference with resteasy reactive - namely that the session is created via a customized handler chain, and the PreMatching filter must be running before this. cc @geoand 

My guess was this is something that at the very least we'd have to document with #27057 (I've added a note about this there), but if it seems like a regression then let's open another issue. 

The / resource must have different handling still (I think we just issue a redirect), so perhaps a session shouldn't be expected there.

Reply:
> The user indicated the PreMatching in prior releases of Keycloak still provided access to the session, so it does appear to be a difference with resteasy reactive - namely that the session is created via a customized handler chain, and the PreMatching filter must be running before this. cc @geoand

I can't say I fully understand the problem, but if this is an issue, we can look into it further together

Reply:
@geoand keycloaksession creation happens here: https://github.com/keycloak/keycloak/blob/77c4cee4d0ea84b17f9ef46fa70638fa16a46c87/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/resteasy/CreateSessionHandler.java#L42

Which is controlled by https://github.com/keycloak/keycloak/blob/77c4cee4d0ea84b17f9ef46fa70638fa16a46c87/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/resteasy/KeycloakHandlerChainCustomizer.java#L62

When a request filter is annotated with PreMatching, what we see is that it the filter will get invoked before the CreateSessionHandler runs.

The user indicates that in prior releases of Keycloak, using just the resteasy classic stack, that a PreMatching filter still had access to the session. 

In the prior releases the keycloaksession creation was handled by QuarkusRequestFilter https://github.com/keycloak/keycloak/commit/217a09ce46bac48b7492a881e2ca5fbeef9dfd6d#diff-80ef3a628be94a94774a6f6e71af1c5d29cfb9dedf6c1e88c39ceb5664662239L134 which was installed via https://github.com/keycloak/keycloak/commit/217a09ce46bac48b7492a881e2ca5fbeef9dfd6d#diff-6ebe7f8db3e4dea9dbefd50eed883b8aac997e6c01127e283631e92e9d022835L614 - which at first glance does appear like it would have priority to run before anything else.

Reply:
I have no idea what the implications would be for Keycloak, but I assume we could the Keycloak handler run before the pre match handler

Reply:
Technically we could create the session at the `BEFORE_PRE_MATCH` phase in the KeycloakHandlerChainCustomizer but I've got no clue why we're not doing it already. Perhaps @pedroigor remembers the reasoning.

Reply:
@vmuzikar I think only because it is the closest method before invoking resources so that we don't have state being held for more time than needed to execute a method in a resource.